x-pack/plugin/core/src/main/resources/fleet-agents.json

```diff
@@ -48,6 +48,19 @@
         "last_checkin_status": {
           "type": "keyword"
         },
+        "last_checkin_message": {
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword",
+              "ignore_above": 1024
+            }
+          }
+        },
```

Comment:
May I ask how last_checking_message is going to be consumed? Is it going to be queried, hence it needs to be tokenized, but also sorted/aggregated on, hence it needs to have a keyword sub-field?

Reply:
I could image both use cases, where it would need to be searched to figure out what or how many agents have particular check-in message. That's why ended up with what seems to be fairly standard auto-mapping for the tex/string content.

Reply:
Cool, it is rather standard but sometimes it is also misused and ends up creating two lucene fields. If the field needs to be tokenized, searched as well as aggregated/sorted, this is the way to go.